File: Regular Expressions/Code/chapter4/4-7/recipe4-7.php
<html>
<head><title>4.7 Limiting user input to alpha characters</title></head>
<style>
	.err { color : red ; font-weight : bold }
</style>
<body>
<form action="recipe4-7.php" method="post">
<input type="text" name="input" 
  value="<? print $_POST['input']; ?>" /><br/>
<input type="submit" value="Submit Form" /><br/><br/>
<?php
if ( $_SERVER['REQUEST_METHOD'] == "POST" ) 
{
	$input = $_POST['input'];	
	if ( preg_match( "/^[A-z]+$/", $input ) )
	{
		# Do some processing here - input if valid
	}
	else
	{
		print "<span class=\"err\">Alpha only.  Please correct and resubmit the form</span><br/>";
	}
}
?>
</form>
</body>
</html>

File: Regular Expressions/Code/chapter4/4-7/recipe4-7.pl
#!/usr/bin/perl -w
use strict;

my $input = $ARGV[0];

if ( $input =~  /^[A-z]+$/ )
{
	print "Thanks for the input:  '" . $input . "'\n";
} else {
	print "Alpha only, please!\n";
}
